tests/core/test_graph.py

```diff
@@ -1,54 +1,367 @@
+    assert out_degrees["a"] == 2
+    assert out_degrees["b"] == 1
+    assert out_degrees["c"] == 0
+
+
+def test_get_laplacian(simple_graph):
+    laplacian = simple_graph.get_laplacian()
+    assert laplacian.shape == (3, 3)
+    assert laplacian.nnz == 6
+    assert isinstance(laplacian, s.coo_matrix)
+    assert laplacian.data[0] == -1
+    assert laplacian.data[1] == -1
+    assert laplacian.data[2] == -1
+    assert laplacian.data[3] == 2
+    assert laplacian.data[4] == 1
+    assert laplacian.data[5] == 0
+
+
+def test_add_node_ohe(simple_graph):
+    simple_graph.add_node_ohe("node_ohe")
+    assert simple_graph.data["node_ohe"].shape == (3, 3)
+    assert torch.equal(simple_graph.data["node_ohe"], torch.eye(3))
+
+    simple_graph.add_node_ohe("node_ohe_dim_small", 2)
+    assert simple_graph.data["node_ohe_dim_small"].shape == (3, 2)
+    assert torch.equal(simple_graph.data["node_ohe_dim_small"], torch.eye(2))
+
+    simple_graph.add_node_ohe("node_ohe_dim_large", 4)
+    assert simple_graph.data["node_ohe_dim_large"].shape == (3, 4)
+    assert torch.equal(simple_graph.data["node_ohe_dim_large"], torch.eye(4))
+
+
+def test_add_edge_ohe(simple_graph):
+    simple_graph.add_edge_ohe("edge_ohe")
+    assert simple_graph.data["edge_ohe"].shape == (3, 3)
+    assert torch.equal(simple_graph.data["edge_ohe"], torch.eye(3))
+
+    simple_graph.add_edge_ohe("edge_ohe_dim_small", 2)
+    assert simple_graph.data["edge_ohe_dim_small"].shape == (3, 2)
+    assert torch.equal(simple_graph.data["edge_ohe_dim_small"], torch.eye(2))
+
+    simple_graph.add_edge_ohe("edge_ohe_dim_large", 4)
+    assert simple_graph.data["edge_ohe_dim_large"].shape == (3, 4)
+    assert torch.equal(simple_graph.data["edge_ohe_dim_large"], torch.eye(4))
+
+
+def test_get_node_attr(simple_graph):
+    simple_graph.data["node_class"] = torch.tensor([[1], [2], [3]])
+    assert simple_graph["node_class"].shape == (3, 1)
+    assert torch.equal(simple_graph["node_class"], torch.tensor([[1], [2], [3]]))
+    assert simple_graph["node_class", "a"].item() == 1
+    assert simple_graph["node_class", "b"].item() == 2
+    assert simple_graph["node_class", "c"].item() == 3
+
+
+def test_get_edge_attr(simple_graph):
+    # Edge indices are sorted during initialization
+    # Potentially leads to wrong weight assignmend
+    simple_graph.data["edge_weight"] = torch.tensor([[1], [1], [2]])
+    assert simple_graph["edge_weight"].shape == (3, 1)
+    assert torch.equal(simple_graph["edge_weight"], torch.tensor([[1], [1], [2]]))
+    assert simple_graph["edge_weight", "a", "b"].item() == 1
+    assert simple_graph["edge_weight", "b", "c"].item() == 1
+    assert simple_graph["edge_weight", "a", "c"].item() == 2
+
+
+def test_get_graph_attr(simple_graph):
+    simple_graph.data["graph_feature"] = torch.tensor([42])
+    assert simple_graph["graph_feature"].item() == 42
+
+
+def test_set_node_attr(simple_graph):
+    simple_graph["node_class"] = torch.tensor([[1], [2], [3]])
```

Comment:
Should assignment like this be possible?

Reply:
Yes, that should be possible.